quixstreams/models/topics/admin.py

```diff
@@ -0,0 +1,194 @@
+        exceptions = {}
+        successful = []
+        existed = []
+        stop_time = time.time() + timeout
+        while futures and time.time() < stop_time:
+            time.sleep(1)
+            for topic_name in list(futures.keys()):
+                future = futures[topic_name]
+                if future.done():
+                    try:
+                        future.result()
+                        successful.append(topic_name)
+                    except KafkaException as e:
+                        # Topic was maybe created by another instance
+                        if e.args[0].name() == "TOPIC_ALREADY_EXISTS":
+                            existed.append(topic_name)
+                        else:
+                            exceptions[topic_name] = e.args[0].str()
+                    # Not sure how these get raised, but they are supposedly possible
+                    except (TypeError, ValueError) as e:
+                        exceptions[topic_name] = e
+                    del futures[topic_name]
+        if successful:
+            logger.info(f"Successfully created topics: {successful}")
+        if existed:
+            logger.info(f"These topics already exist: {existed}")
+        if exceptions:
+            raise CreateTopicFailure(exceptions)
+        if futures:
+            raise CreateTopicTimeout(
+                f"Timed out waiting for creation status for topics:\n"
+                f"{pprint.pformat([topic_name for topic_name in futures])}"
+            )
+
+    def create_topics(
+        self, topics: List[Topic], timeout: int = 10, finalize_timeout: int = 20
+    ):
+        """
+        Create the given list of topics and confirm they are ready.
+
+        Also raises an exception with detailed printout should the creation
+        fail (it ignores issues for a topic already existing).
+
+        :param topics: a list of `Topic`
+        :param timeout: timeout of the creation broker request
+        :param finalize_timeout: the timeout of the topic finalizing ("ready")
+        """
+        existing = self.list_topics().keys()
+        if not (topics := [topic for topic in topics if topic.name not in existing]):
+            logger.info("All topics already exist!")
+            return
+        try:
+            self._finalize_create(
+                self._admin_client.create_topics(
+                    convert_topic_list(topics),
+                    request_timeout=timeout,
+                ),
+                finalize_timeout,
+            )
+        except CreateTopicFailure as e:
+            failure = {
+                topic.name: {
+                    "failure_reason": e.args[0][topic.name],
+                    "topic_config": topic.config.as_dict(),
+                }
+                for topic in topics
+                if topic.name in e.args[0]
+            }
+            raise CreateTopicFailure(
+                f"failed to create topics:\n{pprint.pformat(failure)}"
+            )
```

Comment:
Why do we need to catch and re-raise and exception here vs doing the same thing on line [151](https://github.com/quixio/quix-streams/pull/280/files#diff-fb35e8f5a573400c6f8aa76155325e99b08205dc14b2b9e3d4aa5d9256bd9b2aR151) ?

It looks like the exception is raised only to be re-raised later